Modules/zipimport.c

```diff
@@ -54,46 +54,72 @@ static PyObject *get_data(PyObject *archive, PyObject *toc_entry);
 static PyObject *get_module_code(ZipImporter *self, PyObject *fullname,
                                  int *p_ispackage, PyObject **p_modpath);
 
+static PyTypeObject ZipImporter_Type;
 
 #define ZipImporter_Check(op) PyObject_TypeCheck(op, &ZipImporter_Type)
 
+/*[clinic input]
+module zipimport
+class zipimport.zipimporter "ZipImporter *" "&ZipImporter_Type"
+[clinic start generated code]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=9db8b61557d911e7]*/
+#include "clinic/zipimport.c.h"
+
 
 /* zipimporter.__init__
    Split the "subdirectory" from the Zip archive path, lookup a matching
    entry in sys.path_importer_cache, fetch the file directory from there
    if found, or else read it from the archive. */
+
+/*[clinic input]
+zipimport.zipimporter.__init__
+
+    archivepath: object(converter="PyUnicode_FSDecoder")
```

Comment:
Do you mean `path:` or `archivepath as path:`? 
Sure I can do the latter, but what's the benefit? It was more straight forward to rename the variable to match the signature